src/stores/data-sets.ts

```diff
@@ -93,7 +93,8 @@ export const Datasets = {
   },
 
   getGPSPositionTimeData(name: string, timeRange?: TimeRange) {
-    let dataSet = (PositionTimeDataSets as any)[name] as Dataset;
+    const storeDataSet = (PositionTimeDataSets as any)[name] as Dataset;
+    let dataSet = [...storeDataSet];
```

Comment:
Good point.  This works in practice, but it might be worth using something like `cloneDeep` to add extra safety.

Reply:
In fact, I'm going to switch to that.  Seems like a safer long term option.

Reply:
I was actually thinking this morning that it might be worth looking at something like immer for DESE model state which would eliminate the need for these kinds of shenanigans.